src/components/AddPeopleAndRoles/OrgPerson.vue

```diff
@@ -0,0 +1,533 @@
+    // open confirmation dialog and wait for response
+    this.$refs.reassignCPDialog.open(
+      'Change Completing Party?',
+      this.reassignPersonErrorMessage(),
+      {
+        width: '45rem',
+        persistent: true,
+        yes: 'Change Completing Party',
+        no: null,
+        cancel: 'Cancel'
+      }
+    ).then(async (confirm) => {
+      if (confirm) {
+        this.reassignCompletingParty = true
+      }
+    }).catch(() => {
+      this.isCompletingParty = false
+    })
+  }
+
+  /**
+   * Local helper push the current director data into the list.
+   */
+  private addPerson (): OrgPersonIF {
+    let personToAdd: OrgPersonIF = { ...this.orgPerson }
+    if (this.activeIndex === -1) {
+      personToAdd.id = this.nextId
+    }
+    personToAdd.address = this.setPersonAddress()
+    personToAdd.roles = this.setPersonRoles()
+    return personToAdd
+  }
+
+  private setPersonAddress (): BaseAddressObjIF {
+    let personAddress: BaseAddressObjIF = {
+      mailingAddress: { ...this.inProgressMailingAddress }
+    }
+    if (this.isDirector) {
+      if (this.inheritMailingAddress) {
+        this.inProgressDeliveryAddress = this.inProgressMailingAddress
+      }
+      personAddress = { ...personAddress, deliveryAddress: { ...this.inProgressDeliveryAddress } }
+    }
+    return personAddress
+  }
+
+  private setPersonRoles (): string [] {
+    let roles: string[] = []
+    if (this.isCompletingParty) {
+      roles.push(Roles.COMPLETING_PARTY)
+    }
+    if (this.isDirector) {
+      roles.push(Roles.DIRECTOR)
+    }
+    if (this.isIncorporator) {
+      roles.push(Roles.INCORPORATOR)
+    }
+    return roles
+  }
+
+  private resetAddPersonData (emitEvent: boolean): void {
+    this.$refs.addPersonOrgForm.reset()
+    this.$refs.mailingAddressNew.$refs.addressForm.reset()
+    if (this.$refs.deliveryAddressNew) {
+      this.$refs.deliveryAddressNew.$refs.addressForm.reset()
+    }
+    if (emitEvent) {
+      this.emitResetEvent()
+    }
+  }
+
+  private isRoleLocked (role: string) : boolean {
+    return this.orgPerson.roles.includes(role) && this.activeIndex === -1
+  }
+
+  private removePerson (): void {
+    this.emitRemovePersonEvent(this.activeIndex)
+  }
+
+  private reassignPersonErrorMessage () : string {
+    let errorMessage: string =
+    `<p>The Completing Party role is already assigned to ${this.existingCompletingParty.firstName}
+     ${this.existingCompletingParty.middleName || ''} ${this.existingCompletingParty.lastName}.</p>
+     <p>Selecting "Completing Party" here will change the Completing Party.</p>`
+    return errorMessage
+  }
+
+  get isPerson (): boolean {
+    return this.orgPerson && this.orgPerson.type === IncorporatorTypes.PERSON
+  }
+
+  get isOrg (): boolean {
+    return this.orgPerson && this.orgPerson.type === IncorporatorTypes.CORPORATION
+  }
+
+  get incorporatorLabel () : string {
+    return this.entityFilter(EntityTypes.BCOMP) ? Roles.INCORPORATOR : 'Subscriber'
```

Comment:
Should "Subscriber" be another enum? Seems weird here to use an enum for one value and string for the other value.

Reply:
Sure, makes sense, added!